runtime/src/main/java/org/corfudb/runtime/view/TableRegistry.java

```diff
@@ -664,4 +664,44 @@ public void shutdown() {
             streamingManager.shutdown();
         }
     }
+
+    /**
+     * Populate the ProtobufDescriptorTable using the RegistryTable.
+     * Enables backward compatibility in case of data migration.
+     */
+    public void syncProtobufDescriptorTable() {
```

Comment:
This is pretty hacky and I don't think its a good idea to have it as a public API.

This seems like a build-to-build migration issue, can't it be ignored?

Reply:
This issue can be common for any Corfu upgrade which crosses #2966. Also, it does no harm to call this API even in non-upgrade cases because it only adds protos from RegistryTable to ProtobufDescriptorTable and it doesn't overwrite any existing protos inside ProtobufDescriptorTable. Do you have any better idea to address the issue?

Reply:
I also had the same concern and we had discussed it offline. I also see it like something specific to migration--outside of Corfu... what if we have it as an utility? Or we can have this code sitting in the clients migration code. Cause eventually clients starting from Ufo release would not need it, so it's specific to their workflows.

Reply:
I am leaning on the side of moving this outside corfu.. because from no-one from within Corfu requires it, correct?

Will any third-party client using Corfu require this utility when upgrading from build-to-build?